### PR TITLE
Release/1.4.0

### PR DESCRIPTION
**Pull request by StokesMIDE**
This is the start of the 1.4.0 release.

Created as a draft PR, will be made 'real' after it has been verified that everything we want in the release is in place, and we have a changelist.

### CODE REVIEW

File: endaq/batch/core.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
 from __future__ import annotations
 
-from typing import List
+import typing
+from typing import Any, Dict, List, Callable, Optional, Iterable
```
[the remark this answers is not in the document]
Comment (StokesMIDE) [reply endaq/batch/core.py:4]: > For consistency, might want to refactor to all be `from typing import ...`

If we were to change this, we might want to go the other way. There are a couple of classes in `typing` missing from Python 3.5; they can be referenced in type hinting, but not explicitly imported. I tried changing some things to explicitly import from `typing`, but it caused the tests to fail.

In any case, I don't think we should bother changing it right now.

[the remark this answers is not in the document]
Comment (CFlaniganMide) [reply endaq/batch/core.py:4]: Yeah, it's very minor either way.

[the remark this answers is not in the document]
Comment (CrepeGoat) [reply endaq/batch/core.py:4]: > There are a couple of classes in `typing` missing from Python 3.5 ...

yep, but this affects Python 3.7 as well. A specific example is [`typing.Literal`](https://docs.python.org/3/library/typing.html#typing.Literal), which is 3.8+ only. so using it only works in annotations:
- if you `import typing` and use it as `typing.Literal`
- if you include `from __future__ import annotations` in the file

**Comment by StokesMIDE** [comment]
Type hinting discussion aside, can I get an approval?

**Review by CFlaniganMide — APPROVED** [review]

**Pull request merged**